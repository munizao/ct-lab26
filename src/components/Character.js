import React from 'react';
import PropTypes from 'prop-types';

const Character = (props) => (
  <>
    <div>
      <span>STATUS</span><p>{props.status}</p>
    </div>
    <div>
      <span>SPECIES</span><p>{props.species}</p>
    </div>
    <div>
      <span>GENDER</span><p>{props.gender}</p>
    </div>
    <div>
      <span>ORIGIN</span><p>{props.origin}</p>
    </div>
    <div>
      <span>LAST LOCATION</span><p>{props.lastLocation}</p>
    </div>
  </>
);

Character.propTypes = {
  status: PropTypes.string,
  species: PropTypes.string,
  gender: PropTypes.string,
  origin: PropTypes.string,
  lastLocation: PropTypes.string,
};

export default Character;
