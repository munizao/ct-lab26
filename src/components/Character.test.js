import React from 'react';
import { render } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = render(<Character status="Alive" species="Human" gender="Female" origin="Earth" lastLocation="Mars" />);
    expect(wrapper).toMatchSnapshot();
  });
});