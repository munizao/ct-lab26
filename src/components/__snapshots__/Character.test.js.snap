// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
Array [
  <div>
    <span>
      STATUS
    </span>
    <p>
      Alive
    </p>
  </div>,
  <div>
    <span>
      SPECIES
    </span>
    <p>
      Human
    </p>
  </div>,
  <div>
    <span>
      GENDER
    </span>
    <p>
      Female
    </p>
  </div>,
  <div>
    <span>
      ORIGIN
    </span>
    <p>
      Earth
    </p>
  </div>,
  <div>
    <span>
      LAST LOCATION
    </span>
    <p>
      Mars
    </p>
  </div>,
]
`;
